wallet/src/repository/wallet/state.rs

```diff
@@ -70,10 +70,12 @@ pub struct State {
     pub transaction_next_id: u32,
     /// Current UTXO set (including pending movements)
     pub utxo_set: model::UtxoSet,
+    /// Tagged used UTXO set
+    pub used_outputs: model::UsedOutputs,
```

Comment:
This field should be cleared when resyncing the wallet in `clear_chain_data`.

Reply:
Additionally, this field should be taken into consideration when a revert occurs in the chain.

In other words, in the function `clear_pending_state`, the `usedOutputs` should be reviewed and/or removed (if their timestamp threshold has passed) because they may have some inconsistent state (e.g. an UTXO created in a reverted block).

Reply:
Removing old `usedOutputs` is also encouraged to keep a lower memory footprint`